ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/PortedEndpoints.java

```diff
@@ -113,28 +142,59 @@ void handleTerminateRequest(final RoutingContext routingContext) {
         .apply(requestObject, DefaultApiSecurityContext.create(routingContext));
     completableFuture.thenAccept(endpointResponse -> {
 
-      final Buffer responseBody;
-      try {
-        final byte[] bytes = OBJECT_MAPPER.writeValueAsBytes(endpointResponse.getResponseBody());
-        responseBody = Buffer.buffer(bytes);
-      } catch (JsonProcessingException e) {
-        // This is an internal error as it's a bug in the server
-        routingContext.fail(HttpStatus.SC_INTERNAL_SERVER_ERROR, e);
-        return;
-      }
-
       response.putHeader(CONTENT_TYPE_HEADER, JSON_CONTENT_TYPE);
 
       response.setStatusCode(endpointResponse.getStatusCode())
-          .setStatusMessage(endpointResponse.getStatusMessage())
-          .end(responseBody);
-
+          .setStatusMessage(endpointResponse.getStatusMessage());
+
+      // What the old API returns in it's response is something of a mishmash - sometimes it's
+      // a plain String, other times it's an object that needs to be JSON encoded, other times
+      // it represents a stream.
+      if (endpointResponse instanceof StreamedEndpointResponse) {
+        if (routingContext.request().version() == HttpVersion.HTTP_2) {
+          // The old /query endpoint uses chunked encoding which is not supported in HTTP2
+          routingContext.response().setStatusCode(HttpStatus.SC_METHOD_NOT_ALLOWED)
+              .setStatusMessage("The /query endpoint is not available using HTTP2").end();
```

Comment:
This is because we don't know if the endpoint will return a StreamingOutput (for a print topic or a push query) or just a normal response (for a pull query) until it's been executed. It's just the StreamingOutput (chunked encoding) that's not supported in HTTP2